frontend/src/components/App/App.tsx

```diff
@@ -10,6 +10,16 @@ import { LeftPane } from '../LeftPane/LeftPane'
 import { MapView } from '../MapView/MapView'
 import './App.css'
 
+/**
+ * Typescript complains about implicit type when using import for this package.
+ *
+ * Check to see if the client is mobile and display a warning if they are.
+ */
+const DeviceDetect = require('react-device-detect')
```

Comment:
Don't do this. In types.d.ts, try:

```ts
declare module 'react-device-detect' {
  let detect: {
    isMobile: boolean
  }
 export = detect
}
```

Reply:
See https://www.typescriptlang.org/docs/handbook/modules.html and https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html

Reply:
Ah this is perfect, thanks @bcherny ! I knew I'd need to do something, since otherwise I fall afoul of the `no-var-requires` linter rule :)